### PR TITLE
Documentation: add section about abort dumps

### PR DESCRIPTION
**Pull request by jforissier**

### CODE REVIEW

File: docs/debug.md [hunk 1]
```diff
@@ -287,12 +295,105 @@ graph-time.
...
+ERROR:   [0x0] TEE-CORE:  region 6: va 0 pa 0 size 0 flags ---
+ERROR:   [0x0] TEE-CORE:  region 7: va 0 pa 0 size 0 flags ---
+ERROR:   [0x0] TEE-CORE: Call stack:
+ERROR:   [0x0] TEE-CORE:  0x00105bdc
+ERROR:   [0x0] TEE-CORE:  0x00121bc9
+ERROR:   [0x0] TEE-CORE:  0x00121d1b
+ERROR:   [0x0] TEE-CORE:  0x00121e6f
+```
+
+The above dump was triggered by the TA trying to write to a global 'const'
```
Comment (jbech-linaro) [inline docs/debug.md]: s/'const'/`const`/?

**Comment by jbech-linaro** [comment]
With our without my comment addressed:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: docs/debug.md [hunk 2]
```diff
@@ -287,12 +295,105 @@ graph-time.
...
-# 3. Known issues
-1. Printing the call stack using `bt` makes gdb go into an endless loop.
-   Temporary workaround, in gdb, instead of simply writing `bt`, also mention
-   how many frames you would like to see, for example `bt 10`.
-2. Cannot set breakpoints when the system is up and running. Workaround, set the
-   breakpoints before booting up the system.
+# 3. Abort dumps, panics
+
+When OP-TEE encounters a data or prefetch abort exception, it prints out some
+some diagnostic information to the secure console. The abort may be caused by
```
Comment (etienne-lms) [inline docs/debug.md]: remove `some`

File: docs/debug.md [hunk 3]
```diff
@@ -287,12 +295,105 @@ graph-time.
...
+detected by the TEE core, which can decide to hang the system because there is
+no way to proceed safely. This is called a core panic condition (not to be
+confused with TA panics).
+The messages look slightly different depending on:
+- The error is an exception or a panic,
+- The exception/privilege level when the exception occurred (PL0/EL0 if a
+user mode Trusted Application was running, PL1/EL1 if it was the TEE core) 
+- Whether the TEE and TA are 32 or 64 bits
+- The exact type of exception (data or prefetch abort, translation fault,
+read or write permission fault)
```
Comment (etienne-lms) [inline docs/debug.md]: minor. dots or coma at end of 3 last bullets.

**Comment by jforissier** [comment]
Comments addressed. I have also updated the text to mention TA panics, so this should be merged after https://github.com/OP-TEE/optee_os/pull/1858.

**Comment by jforissier** [comment]
Rebased in the hope that this will fix Travis (no more hello_world repo).

**Comment by jbech-linaro** [comment]
Still looks good, so feel free to use my R-B tag.

**Pull request merged**